src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileValidationServiceImpl.java

```diff
@@ -3,28 +3,169 @@
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
 
 public class ConcurRequestExtractFileValidationServiceImpl implements ConcurRequestExtractFileValidationService {
-
     private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ConcurRequestExtractFileValidationServiceImpl.class);
+    protected ConcurRequestedCashAdvanceService concurRequestedCashAdvanceService;
+    protected ConcurAccountValidationService concurAccountValidationService;
+    protected ParameterService parameterService;
+    protected PersonService personService;
 
-    public boolean requestExtractHeaderRowValidatesToFileContents(List<ConcurRequestExtractFile> requestExtractFiles) {
+    public boolean requestExtractHeaderRowValidatesToFileContents(ConcurRequestExtractFile requestExtractFile) {
         boolean headerValidationPassed = false;
-        if (requestExtractFiles.isEmpty() || requestExtractFiles.size() != 1) {
-            LOG.error("Header validation should have received single parsed file. More or less than one parsed file was detected.");
+        LOG.info("batchDate=" + requestExtractFile.getBatchDate() + "=   recordCount=" + requestExtractFile.getRecordCount() + "=     totalApprovedAmount=" + requestExtractFile.getTotalApprovedAmount() + "=");
+        headerValidationPassed = (fileRowCountMatchesHeaderRowCount(requestExtractFile) &&
+                                  fileOnlyContainsOurEmployeeCustomerIndicator(requestExtractFile) &&
+                                  fileTotalApprovedAmountsMatchHeaderTotalApprovedAmount(requestExtractFile));
+        LOG.info("Header validation : " + ((headerValidationPassed) ? "PASSED" : "FAILED"));
+        return headerValidationPassed;
+    }
+
+    public void performRequestDetailLineValidation(ConcurRequestExtractRequestDetailFileLine detailFileLine) {
+        if (requestDetailFileLineIsCashAdvance(detailFileLine)) {
+            boolean lineValidationPassed = false;
+            lineValidationPassed = (requestedCashAdvanceHasNotBeenUsedInExpenseReport(detailFileLine) &&
+                                    requestedCashAdvanceIsNotBeingDuplicated(detailFileLine) &&
+                                    requestedCashAdvanceAccountingInformationIsValid(detailFileLine));
+            detailFileLine.getValidationResult().setValidCashAdvanceLine(lineValidationPassed);
+            LOG.info("Detail File Line validation : " + ((lineValidationPassed) ? "PASSED" : "FAILED"));
+            LOG.info(KFSConstants.NEWLINE + ((lineValidationPassed) ? KFSConstants.EMPTY_STRING : detailFileLine.toString()));
+        }
+    }
+
+    /**
+     * A request detail line is considered a cash advance when request entry detail type is set to the cash advance indicator.
+     *
+     * @param detailFileLine
+     * @return boolean
+     */
+    private boolean requestDetailFileLineIsCashAdvance(ConcurRequestExtractRequestDetailFileLine detailFileLine) {
```

Comment:
this method name is misleading as this method does not just perform validation and returns a boolean, it is also changing the input object.